# DNA LENGTH

In [ ]:
path= '/content/sequence.fasta'
sequences = []
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        r=len(sequence)
        return r
seq_line(path)

570

In [ ]:
seq="TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT"
r=len(seq)
print(r)

570


### GC%

In [ ]:
path= '/content/sequence.fasta'
sequences = []
count=0
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=list(sequence)
        l=len(seq)
        g=seq.count('G')
        c=seq.count('C')
        count=((g+c)/l)*100
        return count
seq_line(path)

38.59649122807017

### Number of T,G and C

In [ ]:
path= '/content/sequence.fasta'
sequences = []
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=list(sequence)
        print("No. of G in sequence",seq.count('G'))
        print("No. of C in sequence",seq.count('C'))
        print("No. of T in sequence",seq.count('T'))
seq_line(path)

No. of G in sequence 122
No. of C in sequence 98
No. of T in sequence 148


### DNA Single Strand to Double Stranded

In [ ]:
path= '/content/sequence.fasta'
sequences = []
l=[]
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=list(sequence)
        #print(seq)
        for a in range(0,len(seq)):
            if (seq[a]=='A'):
                l.append('T')
            elif (seq[a]=='G'):
                l.append('C')
            elif (seq[a]=='C'):
                l.append('G')
            elif (seq[a]=='T'):
                l.append('A')
        str1 = ''
        for ele in l:
            str1+=ele
        print("1st strand:", sequence)
        print("2nd strand:",str1)
seq_line(path)

1st strand: TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT
2nd strand: AACTAAGTGGTCGCGCATAACAGTAAAGTGTACCATTACCGGGTGTTTAACCTTCAAATAATTAGGTTGTTCGATTTAGCTTGTCACTGTGTTGCGGCCGTTACGGTAACCTATGAACACCACGTTACAGTGTCCCATACTATCCGATAACCAACCTTTGACTTTAGTTAGCGTAAAATTGACTTTACTTATCACTATCTTGACATCCGTGTTAGCAATGTGCACACCTTCATCTATTTCTACTAGGTGCTAAGTTATTGGGTTGGTTTGGTTAACCAGGAAAAATATGCTTTCTTCTTCAACTTCTTAATGTTTTTCTTGTCGGTCTGAGTCAGAAATTTCTTCTACGTCCTGCACCAATATCTTTTCATCAACGCAGTGGTGATGGAGTTAGATATGATCTTG

### Dna to Rna to Protein

In [ ]:
path= '/content/sequence.fasta'
sequences = []
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=sequence
        print("DNA:",seq)
        seq=seq.replace("T","U")
        print("RNA:",seq)
seq_line(path)

DNA: TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT
RNA: UUGAUUCACCAGCGCGUAUUGUCAUUUCACAUGGUAAUGGCCCACAAAUUGGAAGUUUAUUAAUCCAACAAGCUAAAUCGAACAGUGACACAACGCCGGCAAUGCCAUUGGAUACUUGUGGUGCAAUGUCACAGGGUAUGAUAGGCUAUUGGUUGGAAACUGAAAUCAAUCGCAUUUUAACUGAAAUGAAUAGUGAUAGAACUGUAGGCACAAUCGUUACACGUGUGGAAGUAGAUAAAGAUGAUCCACGAUUCAAUAACCCAACCAAACCAAUUGGUCCUUUUUAUACGAAAGAAGAAGUUGAAGAAUUACAAAAAGAACAGCCAGACUCAGUCUUUAAAGAAGAUGCAGGACGUGGUUAUAGAAAAGUAGUUGCGUCACCACUACCUCAAUCUAUACUAGAACACCAGUUAAUUCGA

In [ ]:
#TRANSCRIPTION
path= '/content/sequence.fasta'
sequences = []
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=sequence
        print("DNA:",seq)
        seq=seq.replace("T","A")
        seq=seq.replace("A","U")
        seq=seq.replace("G","C")
        seq=seq.replace("C","G")
        print("RNA:",seq)
seq_line(path)

DNA: TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT
RNA: UUGUUUGUGGUGGGGGUUUUGUGUUUUGUGUUGGUUUUGGGGGUGUUUUUGGUUGUUUUUUUUUGGUUGUUGGUUUUUGGUUGUGUGUGUGUUGGGGGGGUUUGGGUUUGGUUUGUUGUGGUGGUUUGUGUGUGGGUUUGUUUGGGUUUUGGUUGGUUUGUGUUUUGUUUGGGUUUUUUUGUGUUUUGUUUUGUGUUUGUUGUGUUGGGUGUUUGGUUUGUGGUGUGGUUGUUGUUUUUGUUGUUGGUGGUUUGUUUUUGGGUUGGUUUGGUUUUGGUGGUUUUUUUUGGUUUGUUGUUGUUGUUGUUUUUGUUUUUGUUGUGGGUGUGUGUGUGUUUUUUGUUGUUGGUGGUGGUGGUUUUUGUUUUGUUGUUGGGUGUGGUGUUGGUGUUUGUUUUGUUGUUGUGGUGUUUUUUGGU

# OR

In [ ]:
path= '/content/sequence.fasta'
sequences = []
l=[]
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=list(sequence)
        #print(seq)
        for a in range(0,len(seq)):
            if (seq[a]=='A'):
                l.append('U')
            elif (seq[a]=='G'):
                l.append('C')
            elif (seq[a]=='C'):
                l.append('G')
            elif (seq[a]=='T'):
                l.append('A')
        str1 = ''
        for ele in l:
            str1+=ele
        print("1st strand:", sequence)
        print("2nd strand:",str1)
seq_line(path)

1st strand: TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT
2nd strand: AACUAAGUGGUCGCGCAUAACAGUAAAGUGUACCAUUACCGGGUGUUUAACCUUCAAAUAAUUAGGUUGUUCGAUUUAGCUUGUCACUGUGUUGCGGCCGUUACGGUAACCUAUGAACACCACGUUACAGUGUCCCAUACUAUCCGAUAACCAACCUUUGACUUUAGUUAGCGUAAAAUUGACUUUACUUAUCACUAUCUUGACAUCCGUGUUAGCAAUGUGCACACCUUCAUCUAUUUCUACUAGGUGCUAAGUUAUUGGGUUGGUUUGGUUAACCAGGAAAAAUAUGCUUUCUUCUUCAACUUCUUAAUGUUUUUCUUGUCGGUCUGAGUCAGAAAUUUCUUCUACGUCCUGCACCAAUAUCUUUUCAUCAACGCAGUGGUGAUGGAGUUAGAUAUGAUCUUG

In [ ]:
def translate(seq): 
    table = { 
        'AUA':'I', 'AUC':'I', 'AUU':'I', 'AUG':'M', 
        'ACA':'T', 'ACC':'T', 'ACG':'T', 'ACU':'T', 
        'AAC':'N', 'AAU':'N', 'AAA':'K', 'AAG':'K', 
        'AGC':'S', 'AGU':'S', 'AGA':'R', 'AGG':'R',                  
        'CUA':'L', 'CUC':'L', 'CUG':'L', 'CUU':'L', 
        'CCA':'P', 'CCC':'P', 'CCG':'P', 'CCU':'P', 
        'CAC':'H', 'CAU':'H', 'CAA':'Q', 'CAG':'Q', 
        'CGA':'R', 'CGC':'R', 'CGG':'R', 'CGU':'R', 
        'GUA':'V', 'GUC':'V', 'GUG':'V', 'GUU':'V', 
        'GCA':'A', 'GCC':'A', 'GCG':'A', 'GCU':'A', 
        'GAC':'D', 'GAU':'D', 'GAA':'E', 'GAG':'E', 
        'GGA':'G', 'GGC':'G', 'GGG':'G', 'GGU':'G', 
        'UCA':'S', 'UCC':'S', 'UCG':'S', 'UCU':'S', 
        'UUC':'F', 'UUU':'F', 'UUA':'L', 'UUG':'L', 
        'UAC':'Y', 'UAU':'Y', 'UAA':'_', 'UAG':'_', 
        'UGC':'C', 'UGU':'C', 'UGA':'_', 'UGG':'W', 
     } 
    if len(seq)%3 == 0:
        protein = '' 
        for i in range(0, len(seq), 3): 
            codon = seq[i:i + 3] 
            protein+= table[codon] 
    return protein 
path= '/content/sequence.fasta'
sequences = []
l=[]
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=list(sequence)
        #print(seq)
        for a in range(0,len(seq)):
            if (seq[a]=='A'):
                l.append('U')
            elif (seq[a]=='G'):
                l.append('C')
            elif (seq[a]=='C'):
                l.append('G')
            elif (seq[a]=='T'):
                l.append('A')
        str1 = '' 
        for ele in l:
            str1+=ele
        print("1st strand:", sequence)
        print("2nd strand:",str1)
        print("Protein:",translate(str1))
seq_line(path)

1st strand: TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT
2nd strand: AACUAAGUGGUCGCGCAUAACAGUAAAGUGUACCAUUACCGGGUGUUUAACCUUCAAAUAAUUAGGUUGUUCGAUUUAGCUUGUCACUGUGUUGCGGCCGUUACGGUAACCUAUGAACACCACGUUACAGUGUCCCAUACUAUCCGAUAACCAACCUUUGACUUUAGUUAGCGUAAAAUUGACUUUACUUAUCACUAUCUUGACAUCCGUGUUAGCAAUGUGCACACCUUCAUCUAUUUCUACUAGGUGCUAAGUUAUUGGGUUGGUUUGGUUAACCAGGAAAAAUAUGCUUUCUUCUUCAACUUCUUAAUGUUUUUCUUGUCGGUCUGAGUCAGAAAUUUCUUCUACGUCCUGCACCAAUAUCUUUUCAUCAACGCAGUGGUGAUGGAGUUAGAUAUGAUCUUG

Certain amino acids are more or less likely to be found in α-helices or β pleated sheets. For instance, the amino acid proline is sometimes called a “helix breaker” because its unusual R group (which bonds to the amino group to form a ring) creates a bend in the chain and is not compatible with helix formation. Proline is typically found in bends, unstructured regions between secondary structures. Similarly, amino acids such as tryptophan, tyrosine, and phenylalanine, which have large ring structures in their R groups, are often found in β pleated sheets, perhaps because the β pleated sheet structure provides plenty of space for the side chains

### Protein Structure Prediction

In [ ]:
def translate(seq): 
    table = { 
        'AUA':'I', 'AUC':'I', 'AUU':'I', 'AUG':'M', 
        'ACA':'T', 'ACC':'T', 'ACG':'T', 'ACU':'T', 
        'AAC':'N', 'AAU':'N', 'AAA':'K', 'AAG':'K', 
        'AGC':'S', 'AGU':'S', 'AGA':'R', 'AGG':'R',                  
        'CUA':'L', 'CUC':'L', 'CUG':'L', 'CUU':'L', 
        'CCA':'P', 'CCC':'P', 'CCG':'P', 'CCU':'P', 
        'CAC':'H', 'CAU':'H', 'CAA':'Q', 'CAG':'Q', 
        'CGA':'R', 'CGC':'R', 'CGG':'R', 'CGU':'R', 
        'GUA':'V', 'GUC':'V', 'GUG':'V', 'GUU':'V', 
        'GCA':'A', 'GCC':'A', 'GCG':'A', 'GCU':'A', 
        'GAC':'D', 'GAU':'D', 'GAA':'E', 'GAG':'E', 
        'GGA':'G', 'GGC':'G', 'GGG':'G', 'GGU':'G', 
        'UCA':'S', 'UCC':'S', 'UCG':'S', 'UCU':'S', 
        'UUC':'F', 'UUU':'F', 'UUA':'L', 'UUG':'L', 
        'UAC':'Y', 'UAU':'Y', 'UAA':'_', 'UAG':'_', 
        'UGC':'C', 'UGU':'C', 'UGA':'_', 'UGG':'W', 
     } 
    if len(seq)%3 == 0:
        protein = '' 
        for i in range(0, len(seq), 3): 
            codon = seq[i:i + 3] 
            protein+= table[codon] 
    return protein 
path= '/content/sequence.fasta'
sequences = []
l=[]
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('>'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        seq=list(sequence)
        #print(seq)
        for a in range(0,len(seq)):
            if (seq[a]=='A'):
                l.append('U')
            elif (seq[a]=='G'):
                l.append('C')
            elif (seq[a]=='C'):
                l.append('G')
            elif (seq[a]=='T'):
                l.append('A')
        str1 = '' 
        for ele in l:
            str1+=ele
        print("1st strand:", sequence)
        print("2nd strand:",str1)
        print("Protein:",translate(str1))
        protein_list=list(translate(str1))
        #print(protein_list)
        for i in range (0,len(protein_list)):
            if (protein_list[i]=='P'):
                print("Protein Structure: Beta Turn")
                break
            elif (protein_list[i]=='Y'or protein_list[i]=='W' or protein_list[i]=='F'):
                print("Protein Structure: Beta Sheets")
                break
            #else:--------------------------------------------------- I am not sure whether to give this if I give this special cases will be ignored
                #print("Protein Structure: Helix")
                #break
seq_line(path)

1st strand: TTGATTCACCAGCGCGTATTGTCATTTCACATGGTAATGGCCCACAAATTGGAAGTTTATTAATCCAACAAGCTAAATCGAACAGTGACACAACGCCGGCAATGCCATTGGATACTTGTGGTGCAATGTCACAGGGTATGATAGGCTATTGGTTGGAAACTGAAATCAATCGCATTTTAACTGAAATGAATAGTGATAGAACTGTAGGCACAATCGTTACACGTGTGGAAGTAGATAAAGATGATCCACGATTCAATAACCCAACCAAACCAATTGGTCCTTTTTATACGAAAGAAGAAGTTGAAGAATTACAAAAAGAACAGCCAGACTCAGTCTTTAAAGAAGATGCAGGACGTGGTTATAGAAAAGTAGTTGCGTCACCACTACCTCAATCTATACTAGAACACCAGTTAATTCGAACTTTAGCAGACGGTAAAAATATTGTCATTGCATGCGGTGGTGGCGGTATTCCAGTTATAAAAAAAGAAAATACCTATGAAGGTGTTGAAGCGGTTATAGATAAAGATTTTGCTAGTGAGAAATTAGCAACGCTGATTGAAGCAGATACCT
2nd strand: AACUAAGUGGUCGCGCAUAACAGUAAAGUGUACCAUUACCGGGUGUUUAACCUUCAAAUAAUUAGGUUGUUCGAUUUAGCUUGUCACUGUGUUGCGGCCGUUACGGUAACCUAUGAACACCACGUUACAGUGUCCCAUACUAUCCGAUAACCAACCUUUGACUUUAGUUAGCGUAAAAUUGACUUUACUUAUCACUAUCUUGACAUCCGUGUUAGCAAUGUGCACACCUUCAUCUAUUUCUACUAGGUGCUAAGUUAUUGGGUUGGUUUGGUUAACCAGGAAAAAUAUGCUUUCUUCUUCAACUUCUUAAUGUUUUUCUUGUCGGUCUGAGUCAGAAAUUUCUUCUACGUCCUGCACCAAUAUCUUUUCAUCAACGCAGUGGUGAUGGAGUUAGAUAUGAUCUUG

### Protein Parameters

In [ ]:
#Molecular weight of protein
prot="N_VVAHNSKVYHYRVFNLQIIRLFDLACHCVAAVTVTYEHHVTVSHTIR_PTFDFS_RKIDFTYHYLDIRVSNVHTFIYFY_VLSYWVGLVNQEKYAFFFNFLMFFLSV_VRNFFYVLHQYLFINAVVMELDMILWSIKLEIVCHFYNSNVRHHRHKVNIFFFYGYFHNFANIYF_NDHSLIVATNFVYG"
pr=prot.replace("_","")
print(pr)
weights = {'A': 71.04, 'C': 103.01, 'D': 115.03, 'E': 129.04, 'F': 147.07,
       'G': 57.02, 'H': 137.06, 'I': 113.08, 'K': 128.09, 'L': 113.08,
       'M': 131.04, 'N': 114.04, 'P': 97.05, 'Q': 128.06, 'R': 156.10,
       'S': 87.03, 'T': 101.05, 'V': 99.07, 'W': 186.08, 'Y': 163.06,'U':139.11,'O':131.13  }
weight = sum(weights[p] for p in pr)
print ("The molecular weight of this protein is", weight)

NVVAHNSKVYHYRVFNLQIIRLFDLACHCVAAVTVTYEHHVTVSHTIRPTFDFSRKIDFTYHYLDIRVSNVHTFIYFYVLSYWVGLVNQEKYAFFFNFLMFFLSVVRNFFYVLHQYLFINAVVMELDMILWSIKLEIVCHFYNSNVRHHRHKVNIFFFYGYFHNFANIYFNDHSLIVATNFVYG
The molecular weight of this protein is 22318.16000000002


In [ ]:
#Number of amino acids and Isoelectric point of protein
prot="N_VVAHNSKVYHYRVFNLQIIRLFDLACHCVAAVTVTYEHHVTVSHTIR_PTFDFS_RKIDFTYHYLDIRVSNVHTFIYFY_VLSYWVGLVNQEKYAFFFNFLMFFLSV_VRNFFYVLHQYLFINAVVMELDMILWSIKLEIVCHFYNSNVRHHRHKVNIFFFYGYFHNFANIYF_NDHSLIVATNFVYG"
pr=prot.replace("_","")
print(pr)
l=len(pr)
pI={'A':6.00,'R':10.76,'N':5.41,'D':2.77,'C':5.07,'E':3.22,'Q':5.65,'G':5.97,'H':7.59,'I':6.02,'L':5.98,'K':9.74,'M':5.74,'F':5.48,'P':6.30,'U':5.68,'S':5.68,'T':5.60,'W':5.89,'Y':5.66,'V':5.96}
pI = sum(pI[p] for p in pr)
pI=pI/l
print("Number of amino acids",l)
print ("The theoritical PI of this protein is", pI)

NVVAHNSKVYHYRVFNLQIIRLFDLACHCVAAVTVTYEHHVTVSHTIRPTFDFSRKIDFTYHYLDIRVSNVHTFIYFYVLSYWVGLVNQEKYAFFFNFLMFFLSVVRNFFYVLHQYLFINAVVMELDMILWSIKLEIVCHFYNSNVRHHRHKVNIFFFYGYFHNFANIYFNDHSLIVATNFVYG
Number of amino acids 184
The theoritical PI of this protein is 6.063423913043482


The actual protparam output:

Number of amino acids: 184

Molecular weight: 22350.85

Theoretical pI: 8.72

# FASTQ TO FASTA CONVERSION

In [ ]:
path= '/content/seq.fastq'
sequences = []
seq=''
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('@'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        s=sequence.find("+",0,len(sequence))
        seq=sequence[0:s]
        print(">sequence information\n"+seq)
        f = open("demofile2.txt", "a")
        f.write(">sequence information\n"+seq)
        f.close() 
seq_line(path)

>sequence information
GATGAACGCTGGCGGCATGCCTAACACATGCAAGTCGAACGGGGTTTATATGACGGAACTCTCCGGAGGGAAGATATATAAACCTAGTGGCGGACGGGTGAGTAACGCGTGGGCAACCTGCCCTGTACCGGGGGACAACAGCCAGAAATGGCTGCTAATACCGCATAAGCCTGCTGTGCCGCATGGCACAGCAGGGAAAACCCCGGTGGTACAGGATGGGCCCGCGTCGGATTAGTTAGTTGGCAGGGTAACGGCCAGCCAAGCGAACGATCAGGAGCCGGCTGAAGAGAGGGGGCGGCCC


In [ ]:
s="ABCDEFGHIJKLMNOPQRSTUVWXYZ=+GASFDTYINFDSE235781*&"
s.find("+",0,len(s))

27

# QUALITY SCORE SUMMARY

In [ ]:
path= '/content/seq.fastq'
sequences = []
seq=''
def seq_line(path):
    with open(path, 'r') as fin:
        sequence = ''
        for line in fin:
            if line.startswith('@'):
                sequences.append(sequence)
                sequence = ''
            else:
                sequence += line.strip()
        s=sequence.find("+",0,len(sequence))
        l = 0
        ch=''
        for i in range(s+1,len(sequence)):
            ch=ord(sequence[i])
            #print (ch)
            table={'33':0,'34':1,'35':2,'36':3,'37':4,'38':5,'39':6,'40':7,'41':8,'42':9,'43':10,'44':11,'45':12,'46':13,'47':14,'48':15,'49':16,'50':17,'51':18,'52':19,'53':20,'54':21,'55':22,'56':23,'57':24,'58':25,'59':26,'60':27,'61':28,'62':29,'63':30,'64':31,'65':32,'66':33,'67':34,'68':35,'69':36,'70':37,'71':38,'72':39,'73':40}
            for ch in table:
                l += table[ch]
        print(l)
seq_line(path)

246820


# Pandas

In [ ]:
import pandas as pd
#Conversting a simple list into panda series
a=[1,5,6,8,9,6,4,7]
a=pd.Series(a)
a

0    1
1    5
2    6
3    8
4    9
5    6
6    4
7    7
dtype: int64

In [ ]:
type(a)

pandas.core.series.Series

In [ ]:
print("maximum number",a.max())
print("minimum number",a.min())
print("sorted series\n",a.sort_values(ascending=False))

maximum number 9
minimum number 1
sorted series
 4    9
3    8
7    7
2    6
5    6
1    5
6    4
0    1
dtype: int64


In [ ]:
#inplace parameter save the changes made
a.sort_values(ascending=True,inplace=True)
a

0    1
6    4
1    5
5    6
2    6
7    7
3    8
4    9
dtype: int64